packages/daf-w3c/src/action-handler.ts

```diff
@@ -109,54 +114,80 @@ export interface PresentationInput {
   tag?: string
   expirationDate?: string
   verifiableCredential: string[]
+  [x: string]: any
 }
 
 const transformCredentialInput = (input: CredentialInput): VerifiableCredentialPayload => {
-  // TODO validate input
-  const credentialSubject = { ...input.credentialSubject }
-  delete credentialSubject.id
-  const result: VerifiableCredentialPayload = {
-    sub: input.credentialSubject.id,
-    vc: {
-      '@context': input['@context'] || input['context'],
-      type: input.type,
-      credentialSubject,
-    },
-  }
-
-  if (input.expirationDate) {
-    result['exp'] = Date.parse(input.expirationDate) / 1000
-  }
-
-  if (input.id) {
-    result['jti'] = input.id
+  
+  const result = { vc: {} }
+
+  for (const key in input) {
+    switch (key) {
+      case 'credentialSubject':
+        result['sub'] = input[key].id
+        const credentialSubject = { ...input.credentialSubject }
+        delete credentialSubject.id
+        result['vc']['credentialSubject'] = credentialSubject
+      break
+      case '@context':
+      case 'context':
+        result['vc']['@context'] = input[key]
+      break
+      case 'type':
+        result['vc']['type'] = input[key]
+      break
+      case 'expirationDate':
```

Comment:
There should also be a case for  `issuanceDate` that translates to `nbf`
Same transformations about date apply to presentations too and should also be done in reverse when parsing JWT